src/Crypto/JWT.hs

```diff
@@ -227,6 +229,38 @@ instance ToJSON ClaimsSet where
     ] ++ M.toList (filterUnregistered o)
 
 
+-- | Validate the claims made by a ClaimsSet. Currently only inspects
+-- the exp and nbf claims.
+--
+validateClaimsSet
+  :: MonadTime m
+  => State ValidationSettings z
+  -> ClaimsSet
+  -> m Bool
+validateClaimsSet s c =
+  and <$> sequence [ validateExpClaim s c
+                   , validateNbfClaim s c
+                   ]
+
+validateExpClaim
+  :: MonadTime m
+  => State ValidationSettings z
+  -> ClaimsSet
+  -> m Bool
+validateExpClaim _ (ClaimsSet _ _ _ (Just e) _ _ _ _) =
+  (e >) . NumericDate <$> currentTime
+validateExpClaim _ _ = return True
+
+validateNbfClaim
+  :: MonadTime m
+  => State ValidationSettings z
+  -> ClaimsSet
+  -> m Bool
+validateNbfClaim _ (ClaimsSet _ _ _ _ (Just n) _ _ _) =
+  (n <) . NumericDate <$> currentTime
```

Comment:
nbf comparison should be `(n <=)`.  See https://tools.ietf.org/html/rfc7519#section-4.1.5